dozer/cogs/levels.py

```diff
@@ -225,12 +230,21 @@ async def give_message_xp(self, message):
     @command(aliases=["mee6sync"])
     @guild_only()  # Prevent command from being executed in a DM
     @discord.ext.commands.max_concurrency(1, wait=False)  # Only allows one instance of this command to run at a time globally
-    @discord.ext.commands.cooldown(rate=1, per=3600, type=discord.ext.commands.BucketType.guild)  # A cooldown of one hour per guild to prevent spam
+    @discord.ext.commands.cooldown(rate=1, per=900, type=discord.ext.commands.BucketType.guild)  # A cooldown of 15 minutes per guild to prevent spam
     @has_permissions(administrator=True)
     async def meesyncs(self, ctx):
         """Function to scrap ranking data from the mee6 api and save it to the database"""
         guild_id = ctx.guild.id
         progress_template = "Currently syncing from Mee6 API please wait... Page: {page}"
+        DOZER_LOGGER.info(f"Syncing Mee6 level data for {ctx.guild.member_count} members from guild {ctx.guild}({guild_id})")
+
+        results = await GuildXPSettings.get_by(guild_id=int(ctx.guild.id))
+        if len(results):
+            results[0].enabled = False
+
+        await self.sync_to_database()  # We sync the database twice so that the entire cache gets flushed
+        await self.sync_to_database()  # This is to prevent cache entries from overwriting the new synced data
```

Comment:
If a message is received after the first flush and before the syncing is complete, won't the cache still overwrite the database value? A more robust way to do this would be to delete each member's data from the cache just before syncing it.

Reply:
rankings will still show as disabled when the read settings is called, and they can be manually re enabled using the config command